File: Labs/Lab4/Lab4Test.java
import org.junit.Test;

import static org.junit.Assert.*;

public class Lab4Test {
    @Test
    public void test1() {
        TreeNode t = new TreeNode(1);
        t.left = new TreeNode(2);
        t.left.left = new TreeNode(3);
        t.left.right = new TreeNode(4);
        t.right = new TreeNode(8);
        t.right.left = new TreeNode(5);
        t.right.right = new TreeNode(6);

        TreeNode.invertTree(t);

        assertEquals(1, t.val);
        assertEquals(8, t.left.val);
        assertEquals(6, t.left.left.val);
        assertEquals(5, t.left.right.val);
        assertEquals(2, t.right.val);
        assertEquals(4, t.right.left.val);
        assertEquals(3, t.right.right.val);
    }
}

File: Labs/Lab4/lab4Test.py
import unittest
from TreeNode import *

class MyTestCase(unittest.TestCase):
    def test_lab4(self):
        t = TreeNode(1)
        t.left = TreeNode(2)
        t.left.left = TreeNode(3)
        t.left.right = TreeNode(4)
        t.right = TreeNode(8)
        t.right.left = TreeNode(5)
        t.right.right = TreeNode(6)

        t.invertTree(t)

        self.assertEqual(1, t.val)
        self.assertEqual(8, t.left.val)
        self.assertEqual(6, t.left.left.val)
        self.assertEqual(5, t.left.right.val)
        self.assertEqual(2, t.right.val)
        self.assertEqual(4, t.right.left.val)
        self.assertEqual(3, t.right.right.val)



if __name__ == '__main__':
    unittest.main()
